ElixrTalentConnect/Login/LoginViewController.swift

```diff
@@ -0,0 +1,162 @@
+        }
+    }
+        
+        /// Function to get the basic UI layout.
+        func UISetup() {
+            backGroundImage.layer.cornerRadius = 20
+            backGroundImage.clipsToBounds = true
+            signInPromptLabel.text = userInfoPrompt
+            let range = (userInfoPrompt as NSString).range(of: textToChangeColor)
+            let attributedText = NSMutableAttributedString.init(string: userInfoPrompt)
+            attributedText.addAttribute(NSAttributedString.Key.foregroundColor, value: UIColor(red: 228/255, green: 118/255, blue: 75/255, alpha: 1.0) , range: range)
+            signInPromptLabel.attributedText = attributedText
+            let signInGesture = UITapGestureRecognizer(target: self, action:#selector(signUpAction(gesture: ) ))
+            signInPromptLabel.isUserInteractionEnabled = true
+            signInPromptLabel.addGestureRecognizer(signInGesture)
+            
+        }
+        
+        /// Function to setup SignUp action using gesture
+        /// - Parameter gesture: Specific part in the UIlabel where tap gesture is added.
+        @objc func signUpAction(gesture:UITapGestureRecognizer) {
+            let termsRange = (userInfoPrompt as NSString).range(of: self.textToChangeColor)
+            signInPromptLabel.isUserInteractionEnabled = true
+            if gesture.didTapAttributedTextInLabel(label: signInPromptLabel, inRange: termsRange) {
+                print("tapped")
+                guard let rootView = UIStoryboard(name: "Main", bundle: nil).instantiateViewController(withIdentifier: "SignUpViewController") as? SignUpViewController
+                else {
+                    return
+                }
+                navigationController?.pushViewController(rootView, animated: true)
+            }
+        }
+        /// This function is called when the keyboard is about to be displayed.
+        ///It checks the size of the keyboard using information from the notification.
+        /// - Parameter notification: notification on clicking the textfields.
+        @objc func keyboardWillShow(notification: NSNotification) {
+            if let keyboardSize = (notification.userInfo?[UIResponder.keyboardFrameEndUserInfoKey] as? NSValue)?.cgRectValue {
+                if self.view.frame.origin.y == 0 {
+                    self.view.frame.origin.y -= keyboardSize.height
+                }
+            }
+        }
+        
+        /// This implementation is commonly used to handle the keyboard covering the text input fields in a view, ensuring a smooth user experience.
+        /// - Parameter notification: notification on clicking the textfields.
+        @objc func keyboardWillHide(notification: NSNotification) {
+            if self.view.frame.origin.y != 0 {
+                self.view.frame.origin.y = 0
+            }
+        }
+        
+        /// Function  to perform  navigation to Signup viewcontroller.
```

Comment:
Method documentation says the method is to navigate to signup view controller, but the view controller which is getting pushed is LoginViewController. Is this expected?